src/background/messaging/rpc-methods/open.ts

```diff
@@ -13,18 +13,37 @@ import { trackRpcRequestSuccess } from '../rpc-message-handler';
 
 export async function rpcOpen(message: OpenRequest, port: chrome.runtime.Port) {
   const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [['requestId', message.id]]);
-  const { id } = await triggerRequestWindowOpen(RouteUrls.Home, urlParams);
-  void trackRpcRequestSuccess({ endpoint: message.method });
 
-  listenForPopupClose({
-    tabId,
-    id,
-    response: makeRpcErrorResponse('open', {
+  try {
```

Comment:
@kyranjamie : I tried to improve this based on [your feedback](https://github.com/leather-io/extension/issues/5800#issuecomment-2323960176) but I could do with some help.

Now once the window opens I resolve the promise so it's not waiting but I am unsure how the handle the case where the popup is then closed by the user. Maybe that's not necessary?

I based this on `getAddresses`